cdap-cli/src/main/java/co/cask/cdap/cli/CLIConfig.java

```diff
@@ -155,14 +171,19 @@ private AccessToken acquireAccessToken(ClientConfig clientConfig, ConnectionConf
     }
 
     try {
-      AccessToken savedAccessToken = getSavedAccessToken(connectionInfo.getHostname());
-      checkConnection(clientConfig, connectionInfo, savedAccessToken);
-      return savedAccessToken;
+      AccessTokenWrapper savedToken = getSavedAccessToken(connectionInfo.getHostname());
+
+      checkConnection(clientConfig, connectionInfo, savedToken.getAccessToken());
+      connectionInfo.setUsername(savedToken.getUsername());
```

Comment:
username isn't needed for checkConnection. I didn't modify that line anyways in terms on needing a username or not.